docs/cookbook/llm_validation_with_retries.md

```diff
@@ -0,0 +1,155 @@
+# LLM Validation With Retries
+
+!!! info inline end "Concepts Covered"
+
+    - [Calls](ADD LINK)
+    - [Response Model](ADD LINK)
+    - [Tenacity Integration](ADD LINK)
+
+This recipe demonstrates how to leverage Large Language Models (LLMs) -- specifically Anthropic's Claude 3.5 Sonnet -- to perform automated validation on any value. We'll cover how to use **LLMs for complex validation tasks**, how to integrate this with Pydantic's validation system, and how to leverage [Tenacity](ADD LINK) to automatically **reinsert validation errors** back into an LLM call to **improve results**.
+
+!!! note "Background"
+
+    While traditional validation tools like type checkers or Pydantic are limited to hardcoded rules (such as variable types or arithmetic), LLMs allow for much more nuanced and complex validation. This approach can be particularly useful for validating natural language inputs or complex data structures where traditional rule-based validation falls short.
+
+## Basic LLM Validation
+
+Let's start with a simple example of using an LLM to check for spelling and grammatical errors in a text snippet:
+
+```python
+from mirascope.core import anthropic, prompt_template
```

Comment:
for markdowns that refer to a cookbook with multiple .py files, i think adding a comment at the top of each file referencing which .py file the following snippet lives in would be a nice convenience

Reply:
Hmm, I tried playing around with this but didn't like any outcomes for the following reasons:
1. There are some snippets that are broken up, so that becomes weirder to handle
2. The code is already matched to the headers, so the header acts as the file name essentially. I could link the headers, but that looked weird. I've opted instead to add a block at the top of each section, see image below

![Screenshot 2024-08-06 at 11 40 41 AM](https://github.com/user-attachments/assets/937f2874-482a-4eae-aa36-c276141f1691)